packages/api/src/graphql/tag/tag.private-query.ts

```diff
@@ -0,0 +1,95 @@
+import {Prisma, PrismaClient, TagType} from '@prisma/client'
+import {Context} from '../../context'
+import {MaxResultsPerPage} from '../../db/common'
+import {authorise, CanGetTag} from '../permissions'
+import {getSortOrder, SortOrder} from '../queries/sort'
+
+export type TagFilter = {
+  type: TagType
+}
+
+export enum TagSort {
+  CreatedAt = 'CreatedAt',
+  ModifiedAt = 'ModifiedAt'
+}
+
+export const createTagOrder = (
+  field: TagSort,
+  sortOrder: SortOrder
+): Prisma.TagFindManyArgs['orderBy'] => {
+  switch (field) {
+    case TagSort.ModifiedAt:
+      return {
+        modifiedAt: sortOrder
+      }
+
+    case TagSort.CreatedAt:
+    default:
+      return {
+        createdAt: sortOrder
+      }
+  }
+}
+
+const createTypeFilter = (filter?: Partial<TagFilter>): Prisma.TagWhereInput => {
+  if (filter?.type) {
+    return {
+      type: filter?.type
+    }
+  }
+
+  return {}
+}
+
+export const createTagFilter = (filter?: Partial<TagFilter>): Prisma.TagWhereInput => ({
+  AND: [createTypeFilter(filter)]
+})
+
+export const getTags = async (
+  filter: Partial<TagFilter>,
+  sortedField: TagSort,
+  order: 1 | -1,
+  cursorId: string | null,
+  skip: number,
+  take: number,
+  authenticate: Context['authenticate'],
+  tag: PrismaClient['tag']
+) => {
+  const {roles} = authenticate()
+  authorise(CanGetTag, roles)
+
+  const orderBy = createTagOrder(sortedField, getSortOrder(order))
+  const where = createTagFilter(filter)
+
+  const [totalCount, tags] = await Promise.all([
+    tag.count({
+      where: where,
+      orderBy: orderBy
+    }),
+    tag.findMany({
+      where: where,
+      skip: skip,
+      take: Math.min(take, MaxResultsPerPage) + 1,
+      orderBy: orderBy,
+      cursor: cursorId ? {id: cursorId} : undefined
+    })
+  ])
+
+  const nodes = tags.slice(0, take)
```

Comment:
Why is this necessary when you have the take property passed in the findMany() function?

Reply:
Because with cursor based navigation we do not know if there's another page unfortunately